msal/src/main/java/com/microsoft/identity/nativeauth/statemachine/states/AccountState.kt

```diff
@@ -227,42 +230,79 @@ class AccountState private constructor(
      * @return [com.microsoft.identity.nativeauth.statemachine.results.GetAccessTokenResult] The result of the getAccessToken action
      */
     suspend fun getAccessToken(forceRefresh: Boolean = false): GetAccessTokenResult {
+        return getAccessTokenInternal(forceRefresh, null);
+    }
+
+    /**
+     * Retrieves the access token for the currently signed in account from the cache.
+     * If the access token is expired, it will be attempted to be refreshed using the refresh token that's stored in the cache;
+     * Kotlin coroutines variant.
+     *
+     * @return [com.microsoft.identity.nativeauth.statemachine.results.GetAccessTokenResult] The result of the getAccessToken action
+     */
+    suspend fun getAccessToken(forceRefresh: Boolean = false, scopes: List<String>): GetAccessTokenResult {
+        var scopeSet = HashSet<String>()
+        if (!scopes.isNullOrEmpty()) {
+            scopeSet.addAll(scopes)
+        }
+        val scopeSet2 = BaseController.addDefaultScopes(scopeSet)
+        return getAccessTokenInternal(forceRefresh, ArrayList(scopeSet2))
+    }
+
+    suspend fun getAccessTokenInternal(forceRefresh: Boolean, scopes: List<String>?): GetAccessTokenResult {
```

Comment:
should `scopes` be nullable?

Reply:
yes, the scopes are null from the old deprecated method.